### PR TITLE
Support buildOptional and buildEither

### PR DESCRIPTION
**Pull request by Kuniwak**
My motivation is explained in #8.

### CODE REVIEW

File: Tests/AsyncChannelsTests/AsyncChannelTests.swift [hunk 1]
```diff
@@ -266,6 +266,112 @@ final class AsyncTest: XCTestCase {
...
+            }
+        }
+
+        result.close()
+        
+        let r = await result.reduce(into: []) { $0.append($1) }
+        XCTAssertEqual(["foo"], r)
+    }
+    
+    func testOptionalNoneSelect() async throws {
```
Comment (gh123man) [inline Tests/AsyncChannelsTests/AsyncChannelTests.swift:289]: Perhapse a better way to write this test would be to use a `none` case in the select ex:

```swift
func testOptionalNoneSelect() async throws {
    let a = Channel<String>()
    let done = Channel<Bool>()
    
    Task {
        await a <- "foo"
    }
    
    Task {
        await select {
            if false {
                receive(a) { XCTFail() }
            }
            none {
                await done <- true
            }
        }
    }

    await <-done
    a.close()
}
```

This avoids the need for the yield loop and simplifies the logic slightly.

File: Tests/AsyncChannelsTests/AsyncChannelTests.swift [hunk 2]
```diff
@@ -266,6 +266,112 @@ final class AsyncTest: XCTestCase {
...
+        
+        await select {
+            if true {
+                receive(a) { await result <- $0! }
+            }
+        }
+
+        result.close()
+        
+        let r = await result.reduce(into: []) { $0.append($1) }
```
Comment (gh123man) [inline Tests/AsyncChannelsTests/AsyncChannelTests.swift]: For single result asserts, you can use the helper: 
```swift
await assertChanRx(result, "foo")
```
This avoids the need for the reduce statement. 
(Same for other cases below)

**Review by gh123man — CHANGES_REQUESTED** [review]
Thanks for the PR and writing tests - just some very minor comments.

File: Tests/AsyncChannelsTests/AsyncChannelTests.swift (hunk 1 above)
Comment (Kuniwak) [reply Tests/AsyncChannelsTests/AsyncChannelTests.swift:289]: I agree that using `none` is better as you suggested. I fixed it at [f9b1793](https://github.com/gh123man/Async-Channels/pull/10/commits/f9b179350d62c992cb423e1df9d6fa0b1b10a254).

File: Tests/AsyncChannelsTests/AsyncChannelTests.swift (hunk 2 above)
Comment (Kuniwak) [reply Tests/AsyncChannelsTests/AsyncChannelTests.swift]: I fixed it at [80a72f6](https://github.com/gh123man/Async-Channels/pull/10/commits/80a72f671a200fc3b02754d1e9d05049c844c8bc).

**Review by gh123man — APPROVED** [review]
Thanks for the PR! Ill get these published in the next release soon.

**Pull request merged**